### PR TITLE
Adds support for additional consent

### PR DESCRIPTION
**Pull request by patmmccann**
## Type of change

- [ ] Bugfix
- [X] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change
Solves for https://github.com/prebid/Prebid.js/issues/5389 on js following the extension to the TCData object proposed by Google here https://support.google.com/admanager/answer/9681920?hl=en

### CODE REVIEW

**Comment by patmmccann** [comment]
fyi @nielsbaarsma @bretg @bwschmidt

File: modules/consentManagement.js [hunk 1]
```diff
@@ -379,6 +379,13 @@ function storeConsentData(cmpConsentObject) {
       vendorData: (cmpConsentObject) ? cmpConsentObject.getVendorConsents : undefined,
       gdprApplies: (cmpConsentObject) ? cmpConsentObject.getConsentData.gdprApplies : gdprScope
     };
+  } else if (cmpConsentObject.addtlConsent && utils.isStr(cmpConsentObject.addtlConsent)) {
```
Comment (smenzer) [inline modules/consentManagement.js:382]: couldn't this whole `else if` block be replaced with a separate `if` just before the `consentData.apiVersion = cmpVersion;` line that looks like this?

```
if (cmpConsentObject.addtlConsent && utils.isStr(cmpConsentObject.addtlConsent)) {
  consentData.addtlConsent = cmpConsentObject.addtlConsent;
}
```

Comment (patmmccann) [reply modules/consentManagement.js:382]: probably, although that would slightly change the functionality, looking for ac strings on v1 cmps.

Comment (patmmccann) [reply modules/consentManagement.js]: i made a change that preserved the functionality and removed the duplicate conditionals block, with a nested if.

**Comment by bretg** [comment]
docs PR https://github.com/prebid/prebid.github.io/pull/2219

**Comment by patmmccann** [comment]
One example of a cmp supporting this already is https://support.cookiebot.com/hc/en-us/articles/360007652694-Cookiebot-and-the-IAB-Consent-Framework

**Comment by patmmccann** [comment]
From Prebid.js slack, for reviewer: https://prebid.slack.com/archives/C7TLBVAH3/p1597348755293600?thread_ts=1597345850.288600&cid=C7TLBVAH3

**Comment by patmmccann** [comment]
I think I was probably just matching line 639

File: modules/prebidServerBidAdapter/index.js [hunk 2]
```diff
@@ -641,6 +641,9 @@ const OPEN_RTB_PROTOCOL = {
         }
         utils.deepSetValue(request, 'regs.ext.gdpr', gdprApplies);
         utils.deepSetValue(request, 'user.ext.consent', firstBidRequest.gdprConsent.consentString);
+        if (firstBidRequest.gdprConsent.addtlConsent && typeof firstBidRequest.gdprConsent.addtlConsent === 'string') {
+          utils.deepSetValue(request, 'user.ext.ConsentedProvidersSettings.consented_providers', firstBidRequest.gdprConsent.addtlConsent);
```
Comment (jsnellbaker) [inline modules/prebidServerBidAdapter/index.js:645]: Just to confirm - are we sure that this field is supported by PBS?

Comment (patmmccann) [reply modules/prebidServerBidAdapter/index.js:645]: that was the field @bretg proposed on https://github.com/prebid/Prebid.js/issues/5389#issuecomment-669344015 bullet point 2

**Pull request merged**